app/models/plan.rb

```diff
@@ -0,0 +1,27 @@
+# == Schema Information
+#
+# Table name: plans
+#
+#  id                 :bigint           not null, primary key
+#  start_time         :datetime         not null
+#  end_time           :datetime         not null
+#  user_id            :bigint           not null
+#  created_at         :datetime         not null
+#  updated_at         :datetime         not null
+#
+class Plan < ApplicationRecord
+  belongs_to :user, touch: true
+
+  validates :start_time, :end_time, :user_id, presence: true
+  validate :validate_end_time_after_start_time
+
+  def validate_end_time_after_start_time
+    return unless start_time && end_time && end_time < start_time
```

Comment:
```suggestion
    return unless start_time && end_time && end_time <= start_time
```

Reply:
It's quite easy to create a plan with the default start and end which are the same, so perhaps worth validating they set a reasonable end time. (same goes for the time range validation)